### PR TITLE
Remove OSS specific distribution packaging

### PR DESCRIPTION
**Pull request by breskeby**
This removes the OSS projects from distribution projects. Including

- archives
- docker
- packages

This part of elastic/elasticsearch#68797

### CODE REVIEW

**Comment by breskeby** [comment]
@elasticmachine test this please

**Comment by breskeby** [comment]
@elasticmachine test this please

File: qa/wildfly/build.gradle [hunk 1]
```diff
@@ -57,11 +56,7 @@ tasks.named("preProcessFixture").configure {
 }
 
 dockerCompose {
-  if ('default'.equalsIgnoreCase(System.getProperty('tests.distribution', 'oss'))) {
     useComposeFiles = ['docker-compose.yml']
-  } else {
-    useComposeFiles = ['docker-compose-oss.yml']
```
Comment (pugnascotia) [inline qa/wildfly/build.gradle:63]: You can now also remove `qa/wildfly/docker-compose-oss.yml`

File: distribution/docker/build.gradle [hunk 2]
```diff
@@ -149,9 +142,6 @@ project.ext {
           // build-context sub-modules.
           duplicatesStrategy = DuplicatesStrategy.INCLUDE
           from projectDir.resolve("src/docker/config")
-          if (oss) {
-            from projectDir.resolve("src/docker/config/oss")
```
Comment (pugnascotia) [inline distribution/docker/build.gradle:153]: You can remove the directory `distribution/docker/src/docker/config/oss/` as well now.

**Review by pugnascotia — COMMENTED** [review]
Will you change `qa/os/src/test/java/org/elasticsearch/packaging/util/Distribution.java` in a separate PR? It also has a `flavor` field, whose value is derived from the distribution artifact's filename.

**Review by pugnascotia — APPROVED** [review]

**Comment by breskeby** [comment]
@pugnascotia yes that will be another PR. Don't want to mix too much things in one.

**Comment by breskeby** [comment]
@mark-vieira Should this also go into the staged 7.12 branch? By default we only backport build changes to 7.x  ?

**Comment by mark-vieira** [comment]
Only backport to 7.x.

**Comment by mark-vieira** [comment]
Also we should make sure we add the `:Delivery/Packaging` label to stuff like this to ensure that the full packaging test suite CI checks run on these changes in the future.

**Pull request merged**